### PR TITLE
Azure OIDC accounts

### PR DESCRIPTION
**Pull request by benPearce1**
[sc-58116]

PR adds descriptions of main Azure OIDC functionality and refactors the supplied scripts to pull the common bits (create service principal) and specific bits (create password for service principal, create federated creds for service principal)
Adds overview of configuration for OIDC
Adds brief overview of signing keys

### CODE REVIEW

File: src/pages/docs/infrastructure/accounts/azure/index.md [hunk 1]
```diff
@@ -76,25 +97,39 @@ Next, if you want to get even more granular you can constrain the service princi
...
+</details>
+
 <details data-group="infrastructure-accounts-azure">
 <summary>Az PowerShell</summary>
 
 ```powershell
-# This script will create a new service principal for you to use in Octopus Deploy using the Az PowerShell modules.  This will work with both PowerShell and PowerShell Core.
+# this script will create a new Azure AD App Registration
 
 $AzureTenantId = "2a681dca-3230-4e01-abcb-b1fd225c0982" # Replace with your Tenant Id
```
Comment (IsaacCalligeros95) [inline src/pages/docs/infrastructure/accounts/azure/index.md:130]: Nit: Is there a reason this has the tenant specified? This is a pre-existing part of the docs so I'm happy for it to be left as is, the inconsistency is just a bit odd.

File: src/pages/docs/infrastructure/accounts/azure/index.md [hunk 2]
```diff
@@ -112,82 +147,204 @@ Import-Module -Name Az
...
 Write-Host "    2) The Azure Subscription Id: $($azureSubscription.SubscriptionId)"  
-Write-Host "    3) The Azure Application Id: $AzureApplicationId"
-Write-Host "    4) The new password is: $AzurePassword - this is the only time you'll see this password, please store it in a safe location."
-```
+Write-Host "    3) The Azure Application Id: $(AzureApplication.AppId)"
 
+```
 </details>
+
+### Create an Service Principal Client Secret with PowerShell {#create-create-client-secret-via-script}
```
Comment (IsaacCalligeros95) [inline src/pages/docs/infrastructure/accounts/azure/index.md:175]: Double up on create here `#create-create`

File: src/pages/docs/infrastructure/accounts/azure/index.md [hunk 3]
```diff
@@ -112,82 +147,204 @@ Import-Module -Name Az
...
+You can specify the expiry date by adding the *-EndDate* parameter to the *New-AzureRmADApplication* command:
+
+```powershell
+-EndDate (new-object System.DateTime 2018, 12, 31)
+```
+
+Now, you can [add the Service Principal Account in Octopus](#add-service-principal-account). Consider reading our [note on least privilege first](#note_on_least_privilege).
+
+
+### Create an Service Principal Client Secret with PowerShell {#create-create-client-secret-via-script}
```
Comment (IsaacCalligeros95) [inline src/pages/docs/infrastructure/accounts/azure/index.md:265]: Double up on create `#create-create`

**Review by IsaacCalligeros95 — COMMENTED** [review]
Looks good, just a couple of questions/suggestions

* Should we mention the `Octopus.OpenIdConnect.Jwt` variable that gets set and used for token exchange? This could be useful for customers looking to handle their own token exchanges in scripts.

* Can we mention supported tool versions somewhere? In particular for terraform the `azurerm`  terraform provider we require a minimum version of `3.22`. Azure cli added support for [MSAL in 2.30](https://github.com/Azure/azure-cli/pull/19853) which is required for the az login ... --federated-token command

File: src/pages/docs/infrastructure/accounts/azure/index.md (hunk 1 above)
Comment (benPearce1) [reply src/pages/docs/infrastructure/accounts/azure/index.md:130]: This is the Azure Active Directory tenant , not an Octopus one

Comment (IsaacCalligeros95) [reply src/pages/docs/infrastructure/accounts/azure/index.md:130]: 🤦‍♂️

File: src/pages/docs/infrastructure/accounts/openid-connect.md [hunk 4]
```diff
@@ -22,9 +22,13 @@ These must be exposed with anonymous access on HTTPS. Without this, the OpenID C
 
 The hostname of the URL that these two endpoints are available on must either be configured under **Configuration->Nodes->Server Uri** or set as the first ListenPrefix in the server configuration. 
 
+## Authenticating using OpenID Connect with third party services and tools
+
+If you have a third-party service or tool that supports OpenID Connect, you can add any OIDC account variable into your projects variable set and use the `Octopus.OpenIDConnect.Jwt` variable to get access to the request token that can be used for authenticating.
```
Comment (IsaacCalligeros95) [inline src/pages/docs/infrastructure/accounts/openid-connect.md:27]: Sorry I was mistaken before, when using the variables it should be `<AccountName>.Octopus.OpenIDConnect.Jwt`

**Review by IsaacCalligeros95 — APPROVED** [review]
There's just the variable name to update when using project variables, other than that LGTM

File: src/pages/docs/deployments/azure/ase/index.md [hunk 5]
```diff
@@ -28,9 +28,9 @@ Contained in the publish profile is the URI of the deployment endpoint (Kudu) fo
 
 For an external ASE that URI will be publicly accessible (e.g. https://yourapp.scm.aseName.p.azurewebsites.net).
```
Comment (hnrkndrssn) [inline src/pages/docs/deployments/azure/ase/index.md:29]:
```suggestion
For an external ASE that URI will be publicly accessible (e.g. https://your-app.scm.aseName.p.azurewebsites.net).
```

For consistency with the changes below.

**Review by hnrkndrssn — APPROVED** [review]
Minor nit but looks good otherwise.

**Pull request merged**